src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -6,20 +6,27 @@ namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
-    // TODO(ekl) should have a queue per distinct resource type required
     absl::MutexLock lock(&mu_);
-    queued_tasks_.push_back(task_spec);
-    RequestNewWorkerIfNeeded(task_spec);
+    // Required to populate the task spec's scheduling class.
+    const SchedulingClass scheduling_class = task_spec.GetSchedulingClass();
+    auto it = task_queues_.find(scheduling_class);
+    if (it == task_queues_.end()) {
+      task_queues_.emplace(scheduling_class, std::deque<TaskSpecification>(1, task_spec));
+    } else {
+      it->second.push_back(task_spec);
+    }
+    RequestNewWorkerIfNeeded(scheduling_class);
```

Comment:
Since the raylet scheduler also takes into account task dependencies, you'll have to grab the list of ObjectID deps from the args and incorporate it as part of the scheduling class.

Reply:
Done